src/compiler/utilities.ts

```diff
@@ -2083,6 +2083,12 @@ namespace ts {
         return start <= textSpanEnd(span) && end >= span.start;
     }
 
+    export function textSpanIntersectsWith2(start1: number, length1: number, start2: number, length2: number) {
+        let end1 = start1 + length1;
+        let end2 = start2 + length2;
+        return start2 <= end1 && end2 >= start1;

```

Comment:
hrmm.  i could.  but that actually seems harder to read.  With the comparisons now what we're doing is always comparing the second span to the first span.  If i flip so that it's two <= comparisons, then we'll be first comparing the second span to the first, and then the first span to the second.